File: coin-change-2/coin-change-2.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        
        
        
        dp = [0] * (amount+1)
        
        dp[0] = 1
        
        for denom in coins:
            
            for i in range(denom, amount+1):

                
                remain = i - denom
                dp[i] = dp[remain] + dp[i]
                
        return dp[amount]

File: coin-change-2/coin-change-2.java
class Solution {
    public int change(int amount, int[] coins) {
        
        
        int[] dp = new int[amount+1];
        dp[0] = 1;
        
        for (int denom : coins){
            
            for(int i = denom; i < amount + 1; i++){
                int remain = i - denom;
                dp[i] = dp[remain] + dp[i];
            }
        }
        
        
        return dp[amount];
    }
}